drivers/sdp3x/sdp3x.c

```diff
@@ -103,7 +108,7 @@ int sdp3x_init(sdp3x_t *dev, const sdp3x_params_t *params)
 int32_t sdp3x_read_single_temperature(sdp3x_t *dev, uint8_t flags)
 {
     _SDP3x_start_triggered(dev, flags);
-#ifndef SDP3X_PARAM_IRQ_PIN
+#ifndef MODULE_SDP3X_IRQ
```

Comment:
```suggestion
if (!IS_USED(MODULE_SDP3X_IRQ) || dev->params.irq_pin == GPIO_UNDEF)
```

Reply:
If the ifdef is changed to if conditional statement, then the code will unnecessarily compiled even the irq pin is not used.

Also, technically we wouldn't need to put the condition ` dev->params.irq_pin == GPIO_UNDEF` since the code would anyways wait for `DATA_READY_SLEEP_US` with the `xtimer_mutex_lock_timeout()` function without interrupts.

Reply:
`!IS_USED(MODULE_SDP3X_IRQ)` is expanded and resolved in compile time, so the compiler will optimize this part out in the case it's not used.

Reply:
Neat! So basically if the `MODULE_SDP3X_IRQ` is satisfied at compile time, then only the corresponding if conditional statement with `dev->params.irq_pin == GPIO_UNDEF` will be compiled. Am I right ?

Reply:
In the snipped proposed by @benpicco, if the module is not there `!IS_USED(MODULE_SDP3X_IRQ)` expands to `1`, so it would short-circuit the if. The check for the `irq_pin` would then only be compiled when the module is there.

Reply:
If it is only compiled if the if condition is not short circuited, then if I do the following in the init function() at line 91, it should work and get compiled even without the irq module (i.e. periph_gpio_irq not being used, thus gpio_int_int() function being unavailable):
```
    if (IS_USED(MODULE_SDP3X_IRQ) && dev->params.irq_pin != GPIO_UNDEF) {
        mutex_init(&dev->mutex);
        /* lock mutex initially to be unlocked when interrupt is raised */
        mutex_lock(&dev->mutex);
        /* Interrupt set to trigger on falling edge of interrupt pin */
        gpio_init_int(params->irq_pin, GPIO_IN, GPIO_FALLING, _sdp3x_irq_callback,
                     dev);
    }
```

Reply:
Ah, ohkay got it!

Reply:
Fixed